src/main/java/Map.java

```
@@ -84,4 +88,59 @@ public TileType getTileType(Position position) {
 
         return tiles[position.getX()][position.getY()];
     }
+
+    public TileType getTileType(Position position){
+        return getTile(position).getTileType();
+    }
+
+    public String generateHTML(){
+        StringBuilder mapHTML = new StringBuilder();
+
+        URL resource = Game.class.getResource("map_prototype.html");
+        File file = new File(resource.getFile());
+
+        try {
+            String content = new String(Files.readAllBytes(file.toPath()));
+            mapHTML.append(content);
+        }catch (IOException e){
+            e.getMessage();
+            return "ERROR GENERATING HTML";
```

Throw an exception instead of returning a string.